cpp/src/arrow/vendored/ProducerConsumerQueue.h

```diff
@@ -123,7 +123,7 @@ struct ProducerConsumerQueue {
     return false;
   }
 
-  // move the value at the front of the queue to given variable
+  // move (or copy) the value at the front of the queue to given variable
```

Comment:
This was a mistake on my part to add it back in.  I've pulled it back out.  I hadn't realized you had made some changes and I was getting confused during the rebase since I didn't expect any changes and I put this back in by accident.  You are right, it is always a move.  I originally had my logic reversed in my head.